### PR TITLE
Gihanesentimentup

### PR DESCRIPTION
**Pull request by gihanen**

### CODE REVIEW

File: root/backend/src/main/java/com/example/backend/BackendApplication.java [hunk 1]
```diff
@@ -80,7 +80,11 @@ private static String getUserQuery() {
 	}
 
 
+
+	public static JSONObject executeSearch(String queryText, List<IApiHandler> apis, String maxResults, String start, String end) throws Exception {
```
Comment (AndrewMiddletonBC) [inline root/backend/src/main/java/com/example/backend/BackendApplication.java:84]: Why are there two declarations of this method?

File: root/backend/src/main/java/com/example/backend/APIController.java [hunk 2]
```diff
@@ -15,7 +15,11 @@
 public class APIController {
     @CrossOrigin
     @GetMapping("/")
+
+    public Map<String, Object> index(@RequestParam String keyword, @RequestParam boolean twitterChoose, @RequestParam boolean redditChoose, @RequestParam boolean youtubeChoose, @RequestParam String maxResults, @RequestParam String start, @RequestParam String end) throws Exception {
```
Comment (AndrewMiddletonBC) [inline root/backend/src/main/java/com/example/backend/APIController.java:19]: Adding new versions of each method that throw exceptions in a long chain is inferior to just adding some simple exception handling inside your sentiment analysis class.  You should do that.

File: root/backend/pom.xml [hunk 3]
```diff
@@ -5,7 +5,11 @@
 	<parent>
 		<groupId>org.springframework.boot</groupId>
 		<artifactId>spring-boot-starter-parent</artifactId>
+
+		<version>2.6.1-SNAPSHOT</version>
```
Comment (AndrewMiddletonBC) [inline root/backend/pom.xml:9]: Two version declarations.  This also conflicts with the version we need to run on aws elastic beanstalk.

File: root/backend/.gitignore [hunk 4]
```diff
@@ -31,10 +31,11 @@ build/
...
 .vscode/
-
+ gihanesentimentup
 .settings
 bin
 .project
 *.class
 private
 aggregateResults.json
+ main
```
Comment (AndrewMiddletonBC) [inline root/backend/.gitignore:41]: What is this for?

File: root/backend/.gitignore [hunk 5]
```diff
@@ -31,10 +31,11 @@ build/
 
 ### VS Code ###
 .vscode/
-
+ gihanesentimentup
```
Comment (AndrewMiddletonBC) [inline root/backend/.gitignore:34]: What is this for?

File: root/backend/src/main/java/handlers/YoutubeApiHandler.java [hunk 6]
```diff
@@ -26,31 +38,85 @@
...
+        this.limiters.add(new RateLimiter(60, 60000));  // 60 requests per minute; currently unimplemented
+
     }
 
     @Override
     public void requestToken() {
 
+    	
+    	// request a token if sufficient budget and token needed
+    	if (hasRequestBudget(1) ) {
```
Comment (AndrewMiddletonBC) [inline root/backend/src/main/java/handlers/YoutubeApiHandler.java:71]: Out of date, check Thien's latest PR.

File: root/backend/src/main/java/handlers/YoutubeApiHandler.java [hunk 7]
```diff
@@ -26,31 +38,85 @@
...
+    SentimentAnalysis sentimentanalysis= new SentimentAnalysis();
+    
     public YoutubeApiHandler(String key, String user) {
         credentials = new HashMap<>();
         credentials.put("api_key", key);
         credentials.put("user_agent", user);
         this.youtubetoken = null;
+
+        this.token = null;
         this.limiters.add(new RateLimiter(60, 60000));  // 60 requests per minute; currently unimplemented
```
Comment (AndrewMiddletonBC) [inline root/backend/src/main/java/handlers/YoutubeApiHandler.java:51]: Not the correct limiter for youtube I'd wager.  This time implementation is reddit specific.

File: root/backend/src/main/java/util/Credentials.java [hunk 8]
```diff
@@ -10,9 +14,23 @@
...
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.LanguageServiceSettings;
+
 public class Credentials {
 
     private static final String DEFAULT_CREDENTIALS_FILE = "private/credentials.json";
+    private static final String DEFAULT_CREDENTIALS_GOOGLEFILE ="private/socialsentanalysis.json";
+
+
+public class Credentials {
```
Comment (AndrewMiddletonBC) [inline root/backend/src/main/java/util/Credentials.java:30]: Duplicate class declaration.  Why?

File: root/backend/src/main/java/com/example/backend/BackendApplication.java (hunk 1 above)
Comment (gihanen) [reply root/backend/src/main/java/com/example/backend/BackendApplication.java:84]: Sorry, I don't understand what you mean?

File: root/backend/src/main/java/util/Credentials.java (hunk 8 above)
Comment (gihanen) [reply root/backend/src/main/java/util/Credentials.java:30]: I don't know what you mean by duplicate. I only have one class declaration

File: root/backend/src/main/java/handlers/YoutubeApiHandler.java (hunk 7 above)
Comment (gihanen) [reply root/backend/src/main/java/handlers/YoutubeApiHandler.java:51]: i erased it wasn't been used. youtube is using youtubetoken

File: root/backend/.gitignore (hunk 5 above)
Comment (gihanen) [reply root/backend/.gitignore:34]: I dont know why it's here i didn't touch the gitignore. Thats the name of my branch, the one i used to push the code on github

File: root/backend/.gitignore (hunk 4 above)
Comment (gihanen) [reply root/backend/.gitignore:41]: I dont know why it's there. i will erase it

File: root/backend/src/main/java/util/Credentials.java (hunk 8 above)
Comment (AndrewMiddletonBC) [reply root/backend/src/main/java/util/Credentials.java:30]: image

Two class declarations.

**Comment by AndrewMiddletonBC** [comment]
There are several duplicate declarations, and this branch is out of date with Thien's changes to main, causing conflicts.  Please address these issues, then I can test your code and we can merge it.

**Pull request closed without merging**